#2 회원가입 서비스 개발

hamryt opened the pull request:
회원가입 서비스 개발
checkstyle 적용

hamryt committed:
#2 Login 구현

hamryt committed:
Create join service

to make join service I made 6 packages for categorise class.
1. config
(DatabaseConfiguration: setting mybatis configuration,
SecurityJavaConfig: for connecting with mysql )
2.contoller
(UserController: RestApi for Post /users)
3.dto
(UserDto: setting join  user information)
4. exception
(EmailExistedException: when other join user has same email then throw
this class exception
UnexpectedInsertException: when inserting not acted then throw this
class exception)
5. mapper
(UserMapper: mapper for userDto and UserMapper.xml)
6. service
(UserServiceImpl: has business logic about user
UserService: interface for UserServiceImpl)!

hamryt committed:
reate join service

to make join service I made 6 packages for categorise class.
1. config
(DatabaseConfiguration: setting mybatis configuration,
SecurityJavaConfig: for connecting with mysql )
2.contoller
2.contoller
(UserController: RestApi for Post /users)
3.dto
3.dto
(UserDto: setting join  user information)
4. exception
(EmailExistedException: when other join user has same email then throw
this class exception
UnexpectedInsertException: when inserting not acted then throw this
class exception)
5. mapper
(UserMapper: mapper for userDto and UserMapper.xml)
6. service
(UserServiceImpl: has business logic about user
UserService: interface for UserServiceImpl)

hamryt committed:
Merge branch 'develop' into feature/join

hamryt committed:
회원가입 서비스 개발
checkstyle 사용

f-lab-dev commented on helparty/src/main/java/com/hamryt/helparty/config/DatabaseConfiguration.java:
```diff
@@ -0,0 +1,37 @@
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+@Configuration
+@EnableTransactionManagement
```
이거를 달아주면 스프링 내부에는 어떤 변화가 있게될까요?

hamryt replied:
@EnableTransactionManagement 이 어노테이션을 @Configuration 와 같이 추가하면 해당 클래스는 하나의 트랜잭션 처럼 처리됩니다. 따라서 실행도중에 예외가 발생하여 중간에서 작업이 멈출 경우 이전에 있던 작업까지 롤백 해주는 기능을 가능하게 해줍니다.

f-lab-dev replied:
그거는 `@Transactional` 아닐까요??

hamryt replied:
@EnableTransactionManagement가 스프링에게  @Transactional이 있는 메소드를 하나의 트랜잭션으로 감싸라고 알려줍니다.   @EnableTransactionManagement는 트랜잭션을 수행하기 위한 설정이라고 생각합니다

f-lab-dev reviewed (commented):
1. 아직 이전 PR에서 리뷰드린게 고쳐지지 않은 부분이 있네요~
2. 리뷰 내용 수정은 같은 브랜치에 작업하셔서 PUSH하시면 됩니다~ 이렇게 매번 PR을 새로 올리지 않아도 됩니다

hamryt committed:
#2 회원가입 구현
1. UserController클래스와 UserServiceImpl클래스에서 용도에 더 알맞는 @RequiredArgsConstructor 어노테이션으로 변경
생성자 주입을 받으므로 필요한 생성자는 final로 지정되기에 해당 변수를 사용하는 생성자를 만들어주는 @RequiredArgConstructor로 변경
2. UserMapper에서 isExistsEmail 메소드가 null이 리턴되지 않으므로 boolean으로 변경
3. UserServiceImpl클래스에서 회원가입할 때 발생하는 예외 상황을 if (userMapper.insertUser(newUser) == 0)로 그 상황을 특정하는 케이스로 변경
4. SecurityUtil 인터페이스에서 클래스 이름을 확장에 용이하게 범용적인 이름으로 변경

hamryt committed:
UserServiceImpl에서 insertUser할 때 발생하는 예외 조건 변경

f-lab-dev commented on helparty/src/main/java/com/hamryt/helparty/util/SecurityUtil.java:
```diff
@@ -0,0 +1,7 @@
+package com.hamryt.helparty.util;
+
+public interface SecurityUtil {
```
자바에서 유틸클래스란 무엇일까요?

hamryt replied:
Util클래스란 코드에서 반복적이고 공통으로 쓰이는 클래스들을 의미하고 보통 이 클래스들은 따로 분리해내어 재사용성을 높입니다.

f-lab-dev replied:
https://www.mimul.com/blog/oop-alternative-to-utility-classes/

hamryt committed:
util 패키지 삭제
SecurityUtil.class, SecurityUtilImpl.class를 SecurityService.class, SecurityServiceImpl.class로 변경

hamryt committed:
SecurityService를 Encryptor로 변경
SecurityServiceImpl를 Sha256Encryt로 변경

f-lab-dev commented:
멘토링때 리뷰 받았던 내용들 고쳐서 올려주세요~~

hamryt committed:
Sha256Encrypt 필요없는 빈 생성자 삭제
